src/rapids_singlecell/preprocessing/_scrublet/__init__.py

```diff
@@ -243,6 +248,8 @@ def _run_scrublet(ad_obs: AnnData, ad_sim: AnnData | None = None):
 
         return {"obs": ad_obs.obs, "uns": ad_obs.uns["scrublet"]}
 
+    _check_gpu_X(adata.X, allow_dask=True)
+
     if batch_key is not None:
```

Comment:
since 1. the whole function is very spaghetti and 2. you’re now adding yet another indentation level for most of the code, I think you should move this if branch into a helper method.